### ISSUE TITLE
Cannot read properties of undefined (reading 'removeExports')

### ISSUE THREAD

**Issue by L15018105103** [CyanSalt/vite-plugin-unused-code#4]
```
error during build:
TypeError: [vite-plugin-unused-code] Cannot read properties of undefined (reading 'removeExports')
    at C:/project/node_modules/vite-plugin-unused-code/dist/index.cjs:137:35
    at Array.filter (<anonymous>)
```
when I modify config as:
```
import unusedCode from 'vite-plugin-unused-code'

export default {
  plugins: [
    unusedCode({
      detectUnusedExport: false
    }),
  ],
}
```
it works，but can not find unused export code

**Comment by chekrd** [comment]
We are experiencing this issue on Windows machines. macOS is fine.

**Comment by CyanSalt** [comment]
@chekrd Great insight! I believe this is because some package uses the local `path` specification (rather than always using the POSIX specification, as Vite or Rollup do). I'll try to reproduce the issue.

**Comment by CyanSalt** [comment]
Since I don't have a Windows device at the moment, I can only do a black box fix. You can try version `0.1.4`, and welcome feedback whether it is fixed or not.

**Comment by chekrd** [comment]
@CyanSalt The original issue is fixed - thanks a lot for the fast response! However, it seems to be inconsistent, reporting unused files on Windows while correctly not reporting them on macOS.

On macOS, this works as expected and does not report anything:

detectUnusedExport: true,  
detectUnusedFiles: true  

However, for the same codebase, enabling detectUnusedFiles reports false positives on Windows. detectUnusedExport seems to be fine.

Before 0.1.4 only detectUnusedExport had be to disabled.

**Comment by CyanSalt** [comment]
@chekrd I'm not sure I understand what you mean, that on macOS it reports fewer files than on Windows? Can you provide more details, such as whether the additional files reported on Windows are indeed unpacked files, and other characteristics of the file type, etc.

**Comment by CyanSalt** [comment]
@chekrd Sorry, I have found and fixed a very basic bug, maybe it's related to this issue. You can try `0.1.5`.

**Comment by chekrd** [comment]
Now, it reports almost all files in Windows, regardless of whether they are used. I think we need unit tests covering your functions to isolate the source of the issue. Then, I can run them on Windows to help you find it.

Staying on 0.1.3 with unusedExports disabled for now. Versions 0.1.4 and 0.1.5 should be reverted (release 0.1.6 + mark them as unstable) as they are generally worse than 0.1.3.

**Comment by CyanSalt** [comment]
@chekrd You are right, I will add test cases as soon as I have free time and ensure it works under macOS. Thank you for your support!

**Comment by CyanSalt** [comment]
@chekrd I created a very basic test case and confirmed that it works fine on macOS. If what you said is true, this case should not work on Windows. Can you please help test it if possible?

**Comment by chekrd** [comment]
These tests are not very useful, as they are equivalent to running Vite with your plugin. Here is the Windows output:

<img width="686" alt="Image" src="https://github.com/user-attachments/assets/30f78ced-b210-49d2-bf43-6a939a74e44c" />

I doubt there is anything wrong with Vite. The issue we are trying to identify is in your plugin. To isolate it, the tests must specifically cover [your functions](https://github.com/CyanSalt/vite-plugin-unused-code/blob/main/src/index.ts) like getModules, filterGlobs, and others. It means, your expected inputs and outputs of those functions.

I will try to do my best with the current tests though.

**Comment by chekrd** [comment]
I was able to fix the 'unused files' branch in the plugin like this (fast and dirty):

```
/// src/index.ts, unusedCodePlugin function changes:
...
// line 143
const modules = getModules(bundle)
// Get keys from modules, convert win32 segments to posix (converts backslash with /):
const moduleFilePaths = Array.from(modules, ([key]) => key).map((pathname => pathname.split(path.win32.sep).join(path.posix.sep)))
...
if (detectUnusedFiles) {
   unusedFiles = includedFiles.filter(file => !moduleFilePaths.includes(file))
...
```

To fix all issues (unused exports branch), you must convert all `modules` Map values (values, not only keys) with `pathname.split(path.win32.sep).join(path.posix.sep)`

**Comment by CyanSalt** [comment]
Oh, now I understand what you mean, you want to locate the specific function with the problem through "unit" tests. I think it is reasonable, I will try to update it.

**Comment by CyanSalt** [comment]
It sounds because `fast-glob` produces posix-style paths, while rollup provides native-style paths.

**Comment by CyanSalt** [comment]
I have added unit tests for partials of `fast-glob`, `micromatch`, and vite/rollup. You can check which case triggers the issue.

**Comment by chekrd** [comment]
Here is the result:

![Image](https://github.com/user-attachments/assets/4a3394c4-069e-4347-a1f8-5bb7dfc95bf3)

**Comment by CyanSalt** [comment]
@chekrd https://github.com/CyanSalt/vite-plugin-unused-code/commit/3783968584dd50e410f92589f0acb7ff7410152c Maybe this commit fixes the issue. Try running the tests on the latest main branch. I'll release a new version if it works.

**Comment by chekrd** [comment]
All green. I think you can publish and then I can try it on our Windows CI machines.

**Comment by CyanSalt** [comment]
@chekrd [v0.1.6](https://github.com/CyanSalt/vite-plugin-unused-code/releases/tag/v0.1.6) has been released.

**Comment by chekrd** [comment]
All good can be closed, thanks!
